src/plugins/safeSnap/utils/umaModule.ts

```diff
@@ -68,17 +74,90 @@ export const getModuleDetails = async (
     [moduleAddress, 'liveness']
   ]);
   // NOTE: Need to get questionId, finalizedAt, isApproved
+  let needsApproval = false;
+  const minimumBond = moduleDetails[3][0];
+  const optimisticOracle = moduleDetails[1][0];
   const bondDetails = await getBondDetails(provider, moduleAddress);
 
-  const minimumBond = moduleDetails[3][0];
-  let needsApproval = false;
   if (
     Number(minimumBond) > 0 &&
     Number(minimumBond) > Number(bondDetails.currentUserBondAllowance)
   ) {
     needsApproval = true;
   }
 
+  // Create ancillary data for proposal hash
+  let ancillaryData = '';
+  let timestamp = 0;
+  if (transactions !== undefined) {
+    const transactionsHash = keccak256(
+      defaultAbiCoder.encode(
+        ['(address to, uint8 operation, uint256 value, bytes data)[]'],
+        [transactions]
+      )
+    );
+
+    ancillaryData = pack(
+      ['string', 'bytes', 'bytes'],
+      [
+        '',
+        pack(['string', 'string'], ['proposalHash', ':']),
+        toUtf8Bytes(transactionsHash.replace('0x', ''))
+      ]
+    );
+    const moduleContract = new Contract(
+      moduleAddress,
+      UMA_MODULE_ABI,
+      provider
+    );
+    timestamp = await moduleContract.proposalHashes(transactionsHash);
+  }
+
+  // Search for requests with matching ancillary data
+  const oracleContract = new Contract(
+    optimisticOracle,
+    UMA_ORACLE_ABI,
+    provider
+  );
+  const events = await oracleContract.queryFilter('ProposePrice', -100000);
```

Comment:
For now, can we put a TODO here to customize this block lookback based on chain?

Today, I think this will work for mainnet, but I doubt this will work well on L2s (especially polygon) because a) L2 nodes generally enforce a more limited lookback and b) their block times are much faster.

Reply:
Good suggestion, I added a comment to customize the block lookback based on chain and test with L2's (specifically Polygon as it is supported by the Zodiac module).